core/src/main/java/io/confluent/kafka/schemaregistry/rest/MutableHttpServletRequest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2023 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafka.schemaregistry.rest;
+
+
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+
+import org.glassfish.jersey.internal.util.collection.StringKeyIgnoreCaseMultivaluedMap;
+
+final class MutableHttpServletRequest extends HttpServletRequestWrapper {
+  // Allows for adding and replacing custom headers in the HttpServletRequest object.
+  // Only one value per custom header is allowed.
+  private final StringKeyIgnoreCaseMultivaluedMap<String> customHeaders;
+
+  public MutableHttpServletRequest(HttpServletRequest request) {
+    super(request);
+    this.customHeaders = new StringKeyIgnoreCaseMultivaluedMap<String>();
+  }
+
+  public void putHeader(String name, String value) {
+    // Putting a new header will take precedence over existing values in HttpServletRequest.
```

Comment:
Can you check what would be preferable, having the comment describing the function on top(as below) or the current way?
```java
/**
 * Putting a new header will take precedence over existing values in HttpServletRequest.
 */
``` 
The internal comments of the function look good.